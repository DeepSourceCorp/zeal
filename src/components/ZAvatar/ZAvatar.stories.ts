import ZAvatar from './ZAvatar.vue'

export default {
  title: 'Avatar',
  component: ZAvatar,
  excludeStories: /.*Data$/
}

export const AsALink = () => ({
  components: { ZAvatar },
  template: `
    <div class='padded-container'>
      <z-avatar
        type="link"
        image="https://randomuser.me/api/portraits/women/24.jpg"
        href="https://example.com"
      ></z-avatar>
    </div>
    `
})

export const InLoadingState = () => ({
  components: { ZAvatar },
  template: `
    <div class='padded-container'>
      <z-avatar
        :loading="true"
        type="link"
        image="https://randomuser.me/api/portraits/women/24.jpg"
        href="https://example.com"
      ></z-avatar>
    </div>
    `
})


export const WithUserName = () => ({
  components: { ZAvatar },
  template: `
    <div class='padded-container'>
      <z-avatar
        user-name="Akshay Paliwal"
      ></z-avatar>
    </div>
    `
})

export const WithImageAndUserName = () => ({
  components: { ZAvatar },
  template: `<div class='padded-container'>
      <z-avatar
        image="https://randomuser.me/api/portraits/women/24.jpg"
      ></z-avatar>
    </div>`
})

export const WithSizes = () => ({
  components: { ZAvatar },
  template: `
    <div class='bg-dark padded-container'>
      <div class='padded-container'>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="xs"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="sm"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="md"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="lg"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="xl"
        ></z-avatar>
      </div>
      <div class='padded-container bg-vanilla-100'>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="xs"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="sm"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="md"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="lg"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          size="xl"
        ></z-avatar>
      </div>
      <div class='padded-container'>
        <z-avatar
          user-name="Akshay Paliwal"
          size="xs"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="md"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="lg"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="xl"
        ></z-avatar>
      </div>
      <div class='bg-vanilla-100 padded-container'>
        <z-avatar
          user-name="Akshay Paliwal"
          size="xs"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="sm"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="md"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="lg"
        ></z-avatar>
        <z-avatar
          user-name="Akshay Paliwal"
          size="xl"
        ></z-avatar>
      </div>
    </div>
    `
})
