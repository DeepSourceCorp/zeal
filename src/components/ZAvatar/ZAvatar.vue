<template>
  <component v-if="isVisible" :is="TYPES[type]" class="inline-block rounded-full" :class="stroke">
    <span
      v-if="isLoading"
      :class="[`${SIZES[getSize].classes}`]"
      class="flex items-center justify-center text-center rounded-full bg-gradient-skeleton text-vanilla-300"
    >
    </span>
    <template v-else>
      <img
        v-if="this.image"
        :src="image"
        :alt="userName"
        class="rounded-full"
        :class="[`${SIZES[getSize].classes}`]"
        @error.once="setFallbackImage"
      />
      <span
        v-if="!this.image && this.userName"
        :class="[`${SIZES[getSize].classes}`]"
        class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300"
      >
        {{ getUserInitials }}
      </span>
    </template>
  </component>
</template>

<script lang="ts">
import Vue from 'vue'
const SIZES = {
  '2xs': { classes: 'leading-none h-3 w-3 text-xxs', text: '2xs' },
  xs: { classes: 'leading-none h-4 w-4 text-xxs', text: 'xs' },
  sm: { classes: 'leading-none h-6 w-6 text-xs', text: 'sm' },
  md: { classes: 'leading-none h-8 w-8 text-sm', text: 'md' },
  lg: { classes: 'leading-none h-12 w-12 text-lg', text: 'lg' },
  xl: { classes: 'leading-none h-16 w-16 text-2xl', text: 'xl' }
}

const TYPES = {
  link: 'a',
  div: 'div',
  span: 'span'
}

export default Vue.extend({
  name: 'ZAvatar',
  props: {
    type: {
      type: String,
      default: 'link',
      validator: (type) => Object.keys(TYPES).includes(type)
    },
    loading: {
      type: Boolean,
      default: false
    },
    image: String,
    fallbackImage: {
      type: String,
      required: false,
      default: ''
    },
    userName: {
      type: String,
      required: true
    },
    size: {
      type: String,
      default: SIZES.md.text,
      validator: (size) => Object.keys(SIZES).includes(size)
    },
    stroke: {
      type: String,
      required: false,
      default: 'bg-ink-300 p-0.5'
    }
  },
  data() {
    return {
      isVisible: true,
      SIZES,
      TYPES
    }
  },
  computed: {
    isLoading(): boolean {
      const { $options, $props } = this.$parent as Vue
      return $options.name === 'ZAvatarGroup' ? $props.loading : this.loading
    },
    getSize(): string {
      /**
       * Returns `size` property as size of avatar if parent is not ZAvatarGroup,
       * else it returns `size` property of parent as size of avatar.
       */
      const { $options, $props } = this.$parent as Vue
      return $options.name === 'ZAvatarGroup' ? $props.size : this.size
    },
    getUserInitials(): string {
      /**
       * Returns capitalized initials of the user.
       * (Only from the first two words of the userName)
       */
      return this.userName
        ? this.userName
            .split(' ')
            .map((elem) => elem.charAt(0))
            .slice(0, this.size === '2xs' ? 1 : 2) // Make it a single alphabet for the `2xs` size variant
            .join('')
            .toUpperCase()
        : ''
    }
  },
  methods: {
    setFallbackImage(event: ErrorEvent): void {
      if (!this.fallbackImage) return

      const el = event.target as HTMLImageElement
      if (el.src !== this.fallbackImage) {
        el.src = this.fallbackImage
      }
    }
  }
})
</script>
