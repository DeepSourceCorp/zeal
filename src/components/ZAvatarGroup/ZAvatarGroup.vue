<template>
  <div class="flex items-center">
    <slot></slot>
    <span v-if="limitExceededsBy > 0" class="ml-2"> +{{ limitExceededsBy }} </span>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
const SIZES = {
  xs: { text: 'xs', classes: '-ml-3' },
  sm: { text: 'sm', classes: '-ml-3' },
  md: { text: 'md', classes: '-ml-3' },
  lg: { text: 'lg', classes: '-ml-3' },
  xl: { text: 'xl', classes: '-ml-3' }
}

export default Vue.extend({
  name: 'ZAvatarGroup',
  props: {
    size: {
      type: String,
      default: SIZES.md.text,
      validator: (size) => Object.keys(SIZES).includes(size)
    },
    loading: {
      type: Boolean,
      default: false
    },
    limit: Number
  },
  data() {
    return {
      avatars: [] as Vue[]
    }
  },
  computed: {
    limitExceededsBy(): number {
      /**
       * Returns by how much is the number of
       * avatars greater than the limit.
       */
      return this.avatars.length - this.limit
    }
  },
  mounted() {
    this.acquireAvatarsFromChildren()
    this.onlyShowAvatarsUnderLimit()
    this.assignStylingClassesToAvatars()
  },
  methods: {
    acquireAvatarsFromChildren() {
      /**
       * Acquires all the child components which are Avatars,
       * and stores them.
       */
      this.avatars = this.$children.filter(({ $options }) => $options.name === 'ZAvatar')
    },
    onlyShowAvatarsUnderLimit() {
      /**
       * Hides all the avatars which are
       * beyond the limit.
       */
      this.avatars.map(({ $data }, index) => {
        const isLimitExceeded = index > this.limit - 1
        $data.isVisible = !(this.limit && isLimitExceeded)
      })
    },
    assignStylingClassesToAvatars() {
      /**
       * Assigns custom styling classes
       * to all the avatars.
       */
      this.avatars.map(({ $el }, index) => {
        if (index > 0) {
          $el.classList.add(SIZES[this.size].classes)
        }
      })
    }
  }
})
</script>
