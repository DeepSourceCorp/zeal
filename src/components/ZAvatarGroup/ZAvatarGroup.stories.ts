import ZAvatarGroup from './ZAvatarGroup.vue'
import ZAvatar from '../ZAvatar/index'
import { Component } from 'vue'

export default {
  title: 'Avatar Group',
  component: ZAvatarGroup,
  excludeStories: /.*Data$/
}

export const Default = (): Component => ({
  components: { ZAvatarGroup, ZAvatar },
  template: `<div class='padded-container'>
      <z-avatar-group>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
      </z-avatar-group>
    </div>`
})

export const InLoadingState = (): Component => ({
  components: { ZAvatarGroup, ZAvatar },
  template: `
    <div class='padded-container'>
      <z-avatar-group :loading="true">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
      </z-avatar-group>
    </div>
    `
})

export const WithSize = (): Component => ({
  components: { ZAvatarGroup, ZAvatar },
  template: `<div class='padded-container'>
      <z-avatar-group size="xl">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
      </z-avatar-group>
    </div>`
})

export const WithLimit = (): Component => ({
  components: { ZAvatarGroup, ZAvatar },
  template: `
    <div class='padded-container'>
      <z-avatar-group class="p-2 text-vanilla-100 bg-dark" :limit="3">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
      </z-avatar-group>
      <z-avatar-group class="p-2" :limit="3">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
      </z-avatar-group>
    </div>
  `
})

export const WithUserNameOrImageOrSize = (): Component => ({
  components: { ZAvatarGroup, ZAvatar },
  template: `
    <div class='padded-container'>
      <z-avatar-group class="p-2 text-vanilla-100 bg-dark" :limit="6">
        <z-avatar
          user-name="John Doe"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
        user-name="Tom Marvolo Riddle"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
      </z-avatar-group>
      <z-avatar-group class="p-2" :limit="3" size="xl">
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          user-name="Snoop Dogg"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
        <z-avatar
          image="https://randomuser.me/api/portraits/women/24.jpg"
          user-name="Akshay Paliwal"
        ></z-avatar>
      </z-avatar-group>
    </div>
  `
})
