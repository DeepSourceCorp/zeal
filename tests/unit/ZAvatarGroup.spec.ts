import { createLocalVue, mount, Wrapper } from '@vue/test-utils'
import Vue from 'vue'
import { VueConstructor } from 'vue/types/umd'

import ZAvatar from '../../src/components/ZAvatar'
import ZAvatarGroup from '../../src/components/ZAvatarGroup'

describe('Avatar Group component', () => {
  const sizes = ['2xs', 'xs', 'sm', 'md', 'lg', 'xl']

  const localVue: VueConstructor<Vue> = createLocalVue()
  const mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue> = (options = {}) => {
    return mount(ZAvatarGroup, {
      localVue,
      ...options
    })
  }

  it('renders an avatar group when the component is mounted', () => {
    expect(mountFunction({}).html()).toMatchSnapshot()
  })

  it('renders avatar group', () => {
    const mountedInstance = mountFunction({
      components: {
        ZAvatar
      },
      slots: {
        default: [
          `<z-avatar
            user-name="Akshay Paliwal"
          ></z-avatar>`,
          `<z-avatar
            user-name="Akshay Paliwal"
          ></z-avatar>`,
          `<z-avatar
            user-name="Akshay Paliwal"
          ></z-avatar>`,
          `<z-avatar
            user-name="Akshay Paliwal"
          ></z-avatar>`
        ]
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.findAll('span.rounded-full').length).toEqual(4)
  })

  it.each(sizes)('renders avatar group with the size `%s`', size => {
    const mountedInstance = mountFunction({
      components: {
        ZAvatar
      },
      propsData: {
        size
      },
      slots: {
        default: [
          `
            <z-avatar
              user-name="Akshay Paliwal"
            ></z-avatar>
            <z-avatar
              user-name="Snoop Dogg"
            ></z-avatar>
            <z-avatar
              user-name="Akshay Paliwal"
            ></z-avatar>
            <z-avatar
              user-name="Akshay Paliwal"
            ></z-avatar>
            <z-avatar
              user-name="Akshay Paliwal"
            ></z-avatar>
          `
        ]
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
  })
})
