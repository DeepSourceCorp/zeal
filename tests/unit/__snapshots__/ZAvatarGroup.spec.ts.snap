// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Group component renders an avatar group when the component is mounted 1`] = `
<div class="flex items-center">
  <!---->
</div>
`;

exports[`Avatar Group component renders avatar group 1`] = `
<div class="flex items-center"><a class="inline-block rounded-full bg-ink-300 p-0.5">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-8 w-8 text-sm">
      AP
    </span>
  </a><a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-8 w-8 text-sm">
      AP
    </span>
  </a><a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-8 w-8 text-sm">
      AP
    </span>
  </a><a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-8 w-8 text-sm">
      AP
    </span>
  </a>
  <!---->
</div>
`;

exports[`Avatar Group component renders avatar group with the size \`2xs\` 1`] = `
<div class="flex items-center"><a class="inline-block rounded-full bg-ink-300 p-0.5">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-3 w-3 text-xxs">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-3 w-3 text-xxs">
      SD
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-3 w-3 text-xxs">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-3 w-3 text-xxs">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-3 w-3 text-xxs">
      AP
    </span>
  </a>
  <!---->
</div>
`;

exports[`Avatar Group component renders avatar group with the size \`lg\` 1`] = `
<div class="flex items-center"><a class="inline-block rounded-full bg-ink-300 p-0.5">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-12 w-12 text-lg">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-12 w-12 text-lg">
      SD
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-12 w-12 text-lg">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-12 w-12 text-lg">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-12 w-12 text-lg">
      AP
    </span>
  </a>
  <!---->
</div>
`;

exports[`Avatar Group component renders avatar group with the size \`md\` 1`] = `
<div class="flex items-center"><a class="inline-block rounded-full bg-ink-300 p-0.5">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-8 w-8 text-sm">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-8 w-8 text-sm">
      SD
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-8 w-8 text-sm">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-8 w-8 text-sm">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-8 w-8 text-sm">
      AP
    </span>
  </a>
  <!---->
</div>
`;

exports[`Avatar Group component renders avatar group with the size \`sm\` 1`] = `
<div class="flex items-center"><a class="inline-block rounded-full bg-ink-300 p-0.5">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-6 w-6 text-xs">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-6 w-6 text-xs">
      SD
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-6 w-6 text-xs">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-6 w-6 text-xs">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-6 w-6 text-xs">
      AP
    </span>
  </a>
  <!---->
</div>
`;

exports[`Avatar Group component renders avatar group with the size \`xl\` 1`] = `
<div class="flex items-center"><a class="inline-block rounded-full bg-ink-300 p-0.5">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-16 w-16 text-2xl">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-16 w-16 text-2xl">
      SD
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-16 w-16 text-2xl">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-16 w-16 text-2xl">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-16 w-16 text-2xl">
      AP
    </span>
  </a>
  <!---->
</div>
`;

exports[`Avatar Group component renders avatar group with the size \`xs\` 1`] = `
<div class="flex items-center"><a class="inline-block rounded-full bg-ink-300 p-0.5">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-4 w-4 text-xxs">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-4 w-4 text-xxs">
      SD
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-4 w-4 text-xxs">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-4 w-4 text-xxs">
      AP
    </span>
  </a> <a class="inline-block rounded-full bg-ink-300 p-0.5 -ml-3">
    <!----> <span class="flex items-center justify-center text-center rounded-full bg-ink-200 text-vanilla-300 leading-none h-4 w-4 text-xxs">
      AP
    </span>
  </a>
  <!---->
</div>
`;
